api/hypershift/v1beta1/azure.go

```diff
@@ -90,74 +71,116 @@ type AzureNodePoolPlatform struct {
+	// +kubebuilder:validation:XValidation:rule="!self.split('/')[4].endsWith('.')",message="the resourceGroupName in the subnetID must not end with a period (.) character"
+	// +kubebuilder:validation:XValidation:rule=`self.split('/')[8].matches('[a-zA-Z0-9-_\\.]{2,64}')`,message="The vnetName should be between 2 and 64 characters, consisting only of alphanumeric characters, hyphens, underscores and periods"
+	// +kubebuilder:validation:XValidation:rule="!self.split('/')[8].endsWith('.') && !self.split('/')[8].endsWith('-')",message="the vnetName in the subnetID must not end with either a period (.) or hyphen (-) character"
+	// +kubebuilder:validation:XValidation:rule=`self.split('/')[10].matches('[a-zA-Z0-9][a-zA-Z0-9-_\\.]{0,79}')`,message="The subnetName should be between 1 and 80 characters, consisting only of alphanumeric characters, hyphens and underscores and must start with an alphanumeric character"
+	// +kubebuilder:validation:XValidation:rule="!self.split('/')[10].endsWith('.') && !self.split('/')[10].endsWith('-')",message="the subnetName in the subnetID must not end with a period (.) or hyphen (-) character"
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=355
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="SubnetID is immutable"
 	// +kubebuilder:validation:Required
 	SubnetID string `json:"subnetID"`
 
-	// Diagnostics specifies the diagnostics settings for a virtual machine.
+	// diagnostics specifies the diagnostics settings for a virtual machine.
 	// If not specified, then Boot diagnostics will be disabled.
 	// +optional
 	Diagnostics *Diagnostics `json:"diagnostics,omitempty"`
 
-	// MachineIdentityID is a user-assigned identity assigned to the VMs used to authenticate with Azure services. This
-	// field is expected to exist under the same resource group as HostedCluster.Spec.Platform.Azure.ResourceGroupName. This
+	// machineIdentityID is a user-assigned identity assigned to the VMs used to authenticate with Azure services. The
+	// identify is expected to exist under the same resource group as HostedCluster.Spec.Platform.Azure.ResourceGroupName. This
 	// user assigned identity is expected to have the Contributor role assigned to it and scoped to the resource group
 	// under HostedCluster.Spec.Platform.Azure.ResourceGroupName.
 	//
 	// If this field is not supplied, the Service Principal credentials will be written to a file on the disk of each VM
 	// in order to be accessible by the cloud provider; the aforementioned credentials provided are the same ones as
 	// HostedCluster.Spec.Platform.Azure.Credentials. However, this is less secure than using a managed identity.
 	//
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
+	//
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
```

Comment:
I thought we were removing machine identity cc @bryan-cox

Reply:
From the API, yeah we still need to. I can put in a follow up PR if Alberto doesn't want to do that here. This PR is essentially blocking the cert work.

Reply:
yes, let's follow up in different PR for that.